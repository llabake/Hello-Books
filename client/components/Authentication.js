import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import toastMessage from '../helpers/toastMessage';

/**
 * 
 * 
 * @export
 * @param {any} Component to be rendere
 * @returns 
 */
export default function requireAuthentication(Component) {
  /**
   * 
   * 
   * @class Authenticate
   * @extends {Component}
   */
  class Authenticate extends Component {
    static propTypes = {
      authenticated: PropTypes.bool.isRequired,
      dispatch: PropTypes.func.isRequired
    };
  /**
   * 
   * 
   * @returns { js } protected component 
   * @memberof Authenticate
   */
  render() {
      return (
        <div>
          {this.props.authenticated
            ? <Component {...this.props} />
            :
            <Redirect to={{
              pathname: '/signin',
              state: { from: this.props.location }
            }}
            />
          }
          {/* toastMessage(' Please login to access this route', 'failure') */}
          
        </div>
      );
    }
  }

  const mapStateToProps = ({ userReducer }) => ({
    authenticated: userReducer.authenticated,
  });

  return connect(mapStateToProps)(Authenticate);
}
