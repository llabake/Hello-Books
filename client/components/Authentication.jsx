import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import toastMessage from '../helpers/toastMessage';
import { authenticateUser } from '../helpers/utils';

/**
 * 
 * 
 * @export
 * 
 * @param {any} ComposedComponent the child component to be rendered
 * 
 * @returns {void}
 */
export default function requireAuthentication(ComposedComponent) {
  /**
   * 
   * 
   * @class Authenticate
   * 
   * @extends {Component}
   */
  class Authenticate extends Component {
    static propTypes = {
      // authenticated: PropTypes.bool.isRequired,
      dispatch: PropTypes.func.isRequired
    };
    /**
     * 
     * 
     * @param {any} nextProps
     * 
     * @memberof Authenticate
     * 
     * @returns {void}
     */
    componentWillReceiveProps(nextProps) {
      if (!nextProps.isAuthenticated) {
        <Redirect to={{
          pathname: '/signin',
          state: { from: this.props.location }
        }}
        />
      }
    }
    /**
     * 
     * 
     * @returns { object } returns the protected object
     * 
     * @memberof Authenticate
     */
    render() {
      const { isAuthenticated } = authenticateUser()
      return (
        <div>
          {isAuthenticated
            ? <ComposedComponent {...this.props} />
            :
            <Redirect to={{
              pathname: '/signin',
              state: { from: this.props.location }
            }}
            />
          }
          {/* toastMessage(' Please login to access this route', 'failure') */}

        </div>
      );
    }
  }


  return connect(mapStateToProps, null)(Authenticate);
}

export const mapStateToProps = (state) => {
  return {
    error: state.userReducer.error,
    user: state.userReducer.authUser,
    authenticated: state.userReducer.authenticated
  };
};