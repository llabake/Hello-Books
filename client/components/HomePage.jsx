import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux'

import Slider from '../components/common/Slider';
import Carousel from '../components/common/Carousel';
import BookCard from '../components/common/BookCard';
import { fetchPopularBooks } from '../actions/bookAction';

import leanstart from '../media/lean start.jpg';
import ajaxLoader from '../media/ajax-loader.gif';


/**
 * 
 * 
 * @export
 * @class HomePage
 * @extends {Component}
 */
class HomePage extends Component {
  // static propTypes = {
  //   ,
  // }
  /**
   * Creates an instance of HomePage.
   * @param {any} props
   * 
   * @memberof HomePage
   */
  constructor(props) {
    super(props);
    this.state = {};
  }


  /**
   * @returns {Array} most popular books array
   * 
   * @memberof HomePage
   */
  componentWillMount() {
    this.props.fetchPopularBooks();
  }

  /**
   * 
   * @returns {function} initialises the side bav
   * 
   * @memberof HomePage
   */
  // componentDidMount() {
  //   $(".button-collapse").sideNav();
  // }




  /**
  * 
  * 
  * @returns {Object} html
  * 
  * @memberof HomePage
  */
  render() {
    const { popularBooks, loadingPopularBooks } = this.props;
    return (
      <div>
        <Slider />
        <div >
          <h4 className="center-align">
            <span>Most Popular BOOKS</span>
          </h4>
        </div>
        <div className="divider" />
        {loadingPopularBooks ? 
          <div className="center-align" style={{ marginTop: '1em', marginBottom: '1em' }}>
            <img src={ajaxLoader} alt="Loading..." />
          </div> : ''}
        <div className="section">
          <div className="container">
            <div className="row">
              {!loadingPopularBooks ? popularBooks.length ?
                popularBooks.map((book) => {
                  return <div key={book.id}><BookCard book={book} /></div>
                })
                :
                <h5 className="center-align"> No books found </h5>
                : ''}
            </div>
          </div>
        </div>
        <div className="divider" />
        <div className="section similar">
          <h3 className="center-align">
            Based on readers favorites
          </h3>
          <div><Carousel /></div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    popularBooks: state.bookReducer.popularBooks,
    loadingPopularBooks: state.bookReducer.loadingPopularBooks
  };
};


const mapDispatchToProps = (dispatch) => {
  return {
    fetchPopularBooks: () => dispatch(fetchPopularBooks()),
  };
};


HomePage.propTpyes = {
  fetchPopularBooks: PropTypes.func,
  popularBooks: PropTypes.arrayOf(PropTypes.object)
}
export default connect(mapStateToProps, mapDispatchToProps)(HomePage);
