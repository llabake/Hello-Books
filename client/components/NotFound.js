import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from './common/SearchBar';

const NotFound = (props) =>
  <div className="container" style= {{ height: "95vh"}}>
  <div style={{marginLeft: "0px"}}>
    <h4>This is embarassing, something's missing.</h4>
    <b>404: page not found</b>
  </div>
    <div style={{margin: "10 55px 20 35px",}}>
      <SearchBar />
    </div>
    <div >
      <ul style={{"textAlign": "center"}}>
      <li style={{"display": "block"}}><Link to="/">Home</Link>
      </li>
      <li style={{"display": "block"}}><Link to="/allbooks">Books</Link>
      </li>
      <li style={{"display": "block"}}><Link to="/profile">Profile</Link>
      </li>
      </ul>
    </div>
  </div>

export default NotFound;

