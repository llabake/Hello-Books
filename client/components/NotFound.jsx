import React from 'react';
import { Link } from 'react-router-dom';
import SearchBar from './common/SearchBar';

const NotFound = (props) =>
  <div className="container">
    <div style={{ marginLeft: "0px" }} id="not-found">
      <h4>This is embarassing, something's missing.</h4>
      <b>404: page not found</b>
    </div>
    <div>
      <SearchBar />
    </div>
    <div >
      <ul style={{ "textAlign": "center" }}>
        <li style={{ "display": "block" }}><Link to="/">Home</Link>
        </li>
        <li style={{ "display": "block" }}><Link to="/allbooks">Books</Link>
        </li>
        <li style={{ "display": "block" }}><Link to="/profile">Profile</Link>
        </li>
      </ul>
    </div>
  </div>

export default NotFound;

