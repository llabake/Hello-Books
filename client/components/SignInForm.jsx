import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Redirect, Link } from 'react-router-dom';
import { connect } from 'react-redux'

import  { signInUser }  from '../actions/userAction';
import TextInput from '../components/common/TextInput';
import inputValidator from '../helpers/inputValidator'
import { authenticateUser } from '../helpers/utils';

/**
 * 
 * 
 * @export
 * @class SignInForm
 * @extends {Component}
 */
export class SignInForm extends Component {
  /**
   * Creates an instance of SignInForm.
   * @param {any} props 
   * @memberof SignInForm
   */
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errors: {}, 
      isValid: false,
      saving: false
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  /**
   * 
   * 
   * @param {any} nextProps
   *  
   * @returns {Object} User pbject
   * 
   * @memberof SignInForm
   */
  componentWillReceiveProps(nextProps) {
    if (!nextProps.authenticated) {
      this.setState({ saving: false });
    }
  }

  /**
   * 
   * 
   * @param {any} event 
   * @memberof SignUpForm
   * @returns {Object} object containing state
   */
  handleChange (event) {
    this.setState({
      [event.target.name]: event.target.value
    }, () => this.validate());
  }

  

  /**
   * 
   * 
   * @param {any} event 
   * @memberof SignInForm
   * @returns {Object} object containing user detail
   */
  handleSubmit(event) {
    event.preventDefault();
    const userData = this.state;
    if(this.validate()) {
      this.setState({
        error: {},
        saving: true
      });
      this.props.signInUser(userData);
    }
  }
  /**
   * @returns {Object} object containing validation status and corresponding errors
   * 
   * @memberof SignInForm
   */
  validate() {
    const { errors, isValid } = inputValidator.signIn(this.state);
    this.setState({ isValid, errors });
    return isValid;
  }

  /**
   * 
   * 
   * @returns {Object} object containing user detail
   * @memberof SignInForm
   */
  render () {
    const { errors, isValid, saving, } = this.state;
    const { location } = this.props;
    const { isAuthenticated } = authenticateUser()
    const { from } = location.state || { from: { pathname: '/allbooks' } };
    if (isAuthenticated) {
      return (
        <Redirect to={from} />
      );
    }    
    
    return (
      <div>
        <div id="banner">
          <div className="container form-style signin-form">
            <div className="row">
              <div className="row signup-head">  
                <div className="col s12"> 
                  <h5 className="center-align"> Login to HelloBooks Account </h5> 
                </div>
              </div> 
              <div className="col s12 ">
                <div className="row ">
                  <form className="col s12 signup" onSubmit={this.handleSubmit}>      
                    <TextInput
                    id = 'username'
                    type = 'text'
                    icon = 'account_circle'
                    name = 'username'
                    placeholder = 'Username'
                    onChange = {this.handleChange}
                    value = {this.state.username}
                    errors = {errors.username}
                    />
                    <TextInput
                    id = 'password'
                    type = 'password'
                    icon = 'lock'
                    name = 'password'
                    placeholder = 'Password'
                    onChange = {this.handleChange}
                    value = {this.state.password}
                    errors = {errors.password}
                    />
                    <button type="submit" className="waves-effect waves-light btn" disabled= {!isValid || saving}>Sign In</button>
                  </form>
                </div>
              </div>
            </div>
            <div className="terms">         
                <p> Don't have an account? <Link to="/signup">Sign Up Now</Link>  
                </p> 
            </div>
          </div> 
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    error: state.userReducer.error,
    user: state.userReducer.authUser,
    authenticated: state.userReducer.authenticated
  };
};

export const mapDispatchToProps = (dispatch) => {
  return {
    signInUser: (userData) => dispatch(signInUser(userData))
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(SignInForm);

