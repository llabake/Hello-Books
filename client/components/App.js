import React, { Component } from 'react';
import { render } from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import requireAuthentication from './Authentication';
import HomePage from './HomePage';
import SignUpForm from './SignUpForm';
import SignInForm from './SignInForm';
import AllBooks from './main/AllBooks'
import AddBook from './adminDashBoard/AddBook';
import SingleBook from './main/SingleBook';
import AdminDashBoard from './adminDashBoard/AdminDashBoard';
import UserProfilePage from './userDashBoard/UserProfilePage';
import Favorite from './userFavorite/Favorite';
import SearchResult from './common/SearchResult';
import EditProfile from './userDashBoard/EditProfile';
import NotFound from './NotFound';

import '../styles/style.scss';
import Header from './common/Header';
import Footer from './common/Footer';
/**
 * 
 * 
 * @export
 * @class App
 * @extends {Component}
 */
export default class App extends Component {

  componentDidMount() {
    $('.button-collapse').sideNav({
      menuWidth: 250, // Default is 300
      edge: 'left', // Choose the horizontal origin
      draggable: true // Choose whether you can drag to open on touch screens
    }
  );
  // $('.carousel').carousel({
  //   dist: 0, // Perspective zoom. 
  //   shift: 0, // Set the spacing of the center item.
  //   padding: 50, // Set the padding between non center items.
  //   indicators: true, // Show indicators. Default is false
  // });
  // autoplay()
  // function autoplay() {
    // $('.carousel').carousel('next');
    // setTimeout(autoplay, 4500)}

    $('.slider').slider();
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    $('.tooltipped').tooltip({ delay: 50 });
    $('.tabs').tabs();
    $('.tooltipped').tooltip();
  

    // $('.modal').modal();
    // $('select').material_select();
    
  }
  /**
   * 
   * 
   * @returns {object} main App
   * @memberof App
   */
  render () {
    return (
      <Router>
        <div className="body">
        <header>
        <Header />
        </header>
        <main>
        <Switch>
          <Route exact path='/' component={HomePage} />
          <Route exact path='/signup' component={SignUpForm} />
          <Route exact path='/signin' component={SignInForm} />
          <Route exact path='/allbooks' component={requireAuthentication(AllBooks)} />
          <Route exact path='/addbook' component={requireAuthentication(AddBook)} />
          <Route exact path='/admindashboard' component={requireAuthentication(AdminDashBoard)} />
          <Route exact path='/book/:bookId' component={requireAuthentication(SingleBook)} />
          <Route exact path='/profile' component={requireAuthentication(UserProfilePage)} />
          <Route exact path='/favorite' component={requireAuthentication(Favorite)}/>
          <Route exact path='/search/:searchTerm' component={requireAuthentication(SearchResult)}/>
          <Route exact path='/editprofile' component={requireAuthentication(EditProfile)} />
          <Route component={NotFound} />
        </Switch>
        </main>
        <footer>
        <Footer />
        </footer>
        </div>
      </Router>
    );
  }
}





