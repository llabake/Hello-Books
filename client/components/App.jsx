import React, { Component } from 'react';
import { render } from 'react-dom';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import requireAuthentication from './Authentication';
import HomePage from './HomePage';
import SignUpForm from './SignUpForm';
import SignInForm from './SignInForm';
import AllBooks from './main/AllBooks'
import AddBook from './adminDashBoard/AddBook';
import SingleBook from './main/SingleBook';
import AdminDashBoard from './adminDashBoard/AdminDashBoard';
import UserProfilePage from './userDashBoard/UserProfilePage';
import Favorite from './userFavorite/Favorite';
import SearchResult from './common/SearchResult';
import EditProfile from './userDashBoard/EditProfile';
import NotFound from './NotFound';

import '../styles/style.scss';
import Header from './common/Header';
import Footer from './common/Footer';
/**
 * 
 * 
 * @export
 * @class App
 * @extends {Component}
 */
export default class App extends Component {
  /**
   *@returns {void}
   *
   * @memberof App
   */
  componentDidMount() {
    $('.button-collapse').sideNav({
      menuWidth: 250, // Default is 300
      edge: 'left', // Choose the horizontal origin
      draggable: true // Choose whether you can drag to open on touch screens
    }
    );
    $('.slider').slider({
      full_width: true,
    }
    );
    $('.dropdown-button').dropdown({
      belowOrigin: true,
    });
    $('.tooltipped').tooltip({ delay: 50 });
    $('ul.tabs').tabs();
    $('.tooltipped').tooltip({ delay: 50 });
  }

  /**
   * 
   * 
   * @returns {object} main App
   * @memberof App
   */
  render() {
    return (
      <Router>
        <div className="body">
          <header>
            <Header />
          </header>
          <main>
            <Switch>
              <Route exact path='/' component={HomePage} />
              <Route exact path='/signup' component={SignUpForm} />
              <Route exact path='/signin' component={SignInForm} />
              <Route exact path='/allbooks' component={requireAuthentication(AllBooks)} />
              <Route exact path='/addbook' component={requireAuthentication(AddBook)} />
              <Route exact path='/admindashboard' component={requireAuthentication(AdminDashBoard)} />
              <Route exact path='/book/:bookId' component={requireAuthentication(SingleBook)} />
              <Route exact path='/profile' component={requireAuthentication(UserProfilePage)} />
              <Route exact path='/favorite' component={requireAuthentication(Favorite)} />
              <Route exact path='/search/:searchTerm' component={requireAuthentication(SearchResult)} />
              <Route exact path='/editprofile' component={requireAuthentication(EditProfile)} />
              <Route component={NotFound} />
            </Switch>
          </main>
          <footer>
            <Footer />
          </footer>
        </div>
      </Router>
    );
  }
}





