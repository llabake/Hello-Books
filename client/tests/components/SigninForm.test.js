import React from 'react';
import { Redirect } from 'react-router-dom'
import { SignInForm, mapDispatchToProps, mapStateToProps } from '../../components/SignInForm';

function setup() {
  const props = {
    dispatch: jest.fn(),
    location: {
      state: {
        from: {}
      }
    },
    isAuthenticated: false,
    signInUser: jest.fn(),
    handleChange: jest.fn(),
    handleSubmit: jest.fn(),
    errors: {}
  };

  const state = {
    username: 'username',
    password: 'password',
  };
  const wrapper = shallow(<SignInForm {...props} {...state} />);

  const event = {
    preventDefault: jest.fn(),
    target: {
      name: 'username',
      value: 'anothername'
    }
  };

  return {
    wrapper,
    props,
    state,
    event
  };
}

describe("Sign in page", () => {
  test('should render appropriately', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  })

  test('should redirect to previous page or home if user is already authenticated', () => {
    const { wrapper } = setup();
    wrapper.setProps({ isAuthenticated: true });
    expect(wrapper).toMatchSnapshot();
    // expect(wrapper.find("Redirect")).toHaveLength(1);
  });

  test('should return an object with props required by component from store', () => {
    const store = {
      authenticated: false,
      user: {}
    };
    expect(mapStateToProps({ userReducer: {} })).toHaveProperty('authenticated');
    expect(mapStateToProps({ userReducer: {} })).toHaveProperty('user');
  });

  test('should dispatch needed actions', () => {
    const dispatch = jest.fn();
    expect(mapDispatchToProps(dispatch)).toHaveProperty('signInUser')
    const { signInUser } = mapDispatchToProps(dispatch);
    signInUser();
    expect(dispatch).toHaveBeenCalled();
  })

  test('should change state for username', () => {
    const { wrapper, props, event, state } = setup();
    wrapper.find('.signup').simulate('change', event);
    wrapper.instance().handleChange(event)
    expect(wrapper.instance().state.username).toBe('anothername');
  });
})

describe('submit action', () => {
  test('should handle form submission when the form inputs are valid', () => {
    const { wrapper, props, state } = setup();
    const event = {
      preventDefault: jest.fn(),
      target: {}
    }
    wrapper.setState({
      username: 'username',
      password: 'password'
    });
    props.signInUser.mockReset();
    wrapper.instance().handleSubmit(event)
    expect(wrapper.instance().state.username).toBe('username');
    expect(wrapper.instance().state.password).toBe('password');
    expect(props.signInUser).toHaveBeenCalled()
  });

  test(
    'should show validation errors when required form fields are empty',
    () => {
      const { wrapper, props, state } = setup();
      const event = {
        preventDefault: jest.fn(),
        target: {
          name: 'username',
          value: ''
        }
      };
      wrapper.setState({
        username: '',
        password: ''
      });
      props.signInUser.mockReset();
      wrapper.instance().handleSubmit(event)
      expect(wrapper.instance().state.username).toBe('');
      expect(wrapper.instance().state.password).toBe('');
      expect(wrapper.instance().state.errors.username)
        .toEqual([ 'username is required' ]);
      expect(wrapper.instance().state.errors.password)
        .toEqual([ 'password is required' ]);
    }
  );
});