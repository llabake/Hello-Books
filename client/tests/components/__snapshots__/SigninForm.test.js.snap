// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign in page should redirect to previous page or home if user is already authenticated 1`] = `
<div>
  <div
    id="banner"
  >
    <div
      className="container form-style signin-form"
    >
      <div
        className="row"
      >
        <div
          className="row signup-head"
        >
          <div
            className="col s12"
          >
            <h5
              className="center-align"
            >
               Login to HelloBooks Account 
            </h5>
          </div>
        </div>
        <div
          className="col s12 "
        >
          <div
            className="row "
          >
            <form
              className="col s12 signup"
              onSubmit={[Function]}
            >
              <TextInput
                icon="account_circle"
                id="username"
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />
              <TextInput
                icon="lock"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <button
                className="waves-effect waves-light btn"
                disabled={true}
                type="submit"
              >
                Sign In
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        className="terms"
      >
        <p>
           Don't have an account? 
          <Link
            replace={false}
            to="/signup"
          >
            Sign Up Now
          </Link>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Sign in page should render appropriately 1`] = `
<div>
  <div
    id="banner"
  >
    <div
      className="container form-style signin-form"
    >
      <div
        className="row"
      >
        <div
          className="row signup-head"
        >
          <div
            className="col s12"
          >
            <h5
              className="center-align"
            >
               Login to HelloBooks Account 
            </h5>
          </div>
        </div>
        <div
          className="col s12 "
        >
          <div
            className="row "
          >
            <form
              className="col s12 signup"
              onSubmit={[Function]}
            >
              <TextInput
                icon="account_circle"
                id="username"
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type="text"
                value=""
              />
              <TextInput
                icon="lock"
                id="password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <button
                className="waves-effect waves-light btn"
                disabled={true}
                type="submit"
              >
                Sign In
              </button>
            </form>
          </div>
        </div>
      </div>
      <div
        className="terms"
      >
        <p>
           Don't have an account? 
          <Link
            replace={false}
            to="/signup"
          >
            Sign Up Now
          </Link>
        </p>
      </div>
    </div>
  </div>
</div>
`;
